compiler/commands.nim

```diff
@@ -439,6 +439,13 @@ proc setCmd*(conf: ConfigRef, cmd: Command) =
 proc setCommandEarly*(conf: ConfigRef, command: string) =
   conf.command = command
   setCmd(conf, command.parseCommand)
+  # command early customizations
+  # must be handled here to honor subsequent `--hint:x:on|off`
+  case conf.cmd
+  of cmdRst2html, cmdRst2tex: # xxx see whether to add others: cmdGendepend, etc.
+    conf.foreignPackageNotes = {hintSuccessX}
+  else:
+    conf.foreignPackageNotes = foreignPackageNotesDefault
```

Comment:
What's effect of else parts?

Reply:
it handles the case where `setCommandEarly` is called more than once (eg via nimscript's `setCommand` or cmdline); arguably it's an edge case but `setCommand` is supported